app/lib/features/spaces/providers/space_list_provider.dart

```diff
@@ -22,21 +26,48 @@ final subSpacesListProvider =
   return subSpacesList;
 });
 
-final spaceListSearchProvider = FutureProvider.autoDispose
-    .family<List<Space>, String>((ref, searchText) async {
+final allSpaceListWithBookmarkFrontProvider =
+    Provider.autoDispose<List<Space>>((ref) {
   final bookmarkedSpaceList = ref.watch(bookmarkedSpacesProvider);
   final othersSpaceList = ref.watch(unbookmarkedSpacesProvider);
   final spaceList = bookmarkedSpaceList.followedBy(othersSpaceList);
+  return spaceList.toList();
+});
+
+final spaceListSearchProvider = Provider.autoDispose<List<Space>>((ref) {
+  final spaceList = ref.watch(allSpaceListWithBookmarkFrontProvider);
+  final searchTerm =
+      ref.watch(spaceListSearchTermProvider).trim().toLowerCase();
+
+  //Return all spaces if search is empty
+  final searchValue = searchTerm.trim().toLowerCase();
+  if (searchValue.isEmpty) return spaceList;
+
+  //Return all spaces with search criteria
+  var spacesSearchedList = spaceList.where((space) {
+    final roomId = space.getRoomIdStr();
+    final spaceInfo = ref.watch(roomAvatarInfoProvider(roomId));
```

Comment:
in the past I had noticed that `ref.watch` wouldn't always fully trigger in certain nesting conditions. I believe due them being done in a scope that is lost and thus the ref-watch being removed again. Would be great to have a test for this provider ensure that if the `roomAvatarInfoProvider`-info of an item changes the spaceListSearchProvider properly picks up upon that and shows it in the list.

In the past, this lead to not noticing updates that came in async as being things that are properly matched for the search. It would essentially never refresh displayName-data and if it wasn't available at first try users (in that case back then it was mentions) would never match ...